source/uwp/Renderer/lib/XamlBuilder.cpp

```diff
@@ -1424,75 +1424,78 @@ namespace AdaptiveNamespace
-                                                                           buttonFrameworkElement.Get()));
+                            THROW_IF_FAILED(buttonFrameworkElement->put_Style(styleToApply.Get()));
                         }
-                        else
+                    }
+                }
+                else if (SUCCEEDED(WindowsCompareStringOrdinal(actionSentiment.Get(), HStringReference(L"destructive").Get(), &isSentimentDestructive))
+                         && (isSentimentDestructive == 0))
+                {
+                    if (SUCCEEDED(TryGetResourceFromResourceDictionaries<IStyle>(resourceDictionary.Get(),
+                        L"Adaptive.Action.Destructive",
+                        &styleToApply)))
+                    {
+                        THROW_IF_FAILED(buttonFrameworkElement->put_Style(styleToApply.Get()));
+                    }
+                    else
+                    {
+                        // By default, set the action text color to attention color
+                        ComPtr<IResourceDictionary> actionSentimentDictionary =
+                            contextImpl->GetDefaultActionSentimentDictionary();
+
+                        if (SUCCEEDED(TryGetResourceFromResourceDictionaries(actionSentimentDictionary.Get(),
+                            L"DestructiveActionDefaultStyle",
+                            styleToApply.GetAddressOf())))
                         {
-                            // By default, set the action text color to attention color
-                            ComPtr<IResourceDictionary> actionSentimentDictionary =
-                                contextImpl->GetDefaultActionSentimentDictionary();
-
-                            ComPtr<IStyle> actionDestructiveSentimentStyle;
-                            if (SUCCEEDED(TryGetResourceFromResourceDictionaries(actionSentimentDictionary.Get(),
-                                                                                 L"DestructiveActionDefaultStyle",
-                                                                                 actionDestructiveSentimentStyle.GetAddressOf())))
-                            {
-                                THROW_IF_FAILED(buttonFrameworkElement->put_Style(actionDestructiveSentimentStyle.Get()));
-                            }
+                            THROW_IF_FAILED(buttonFrameworkElement->put_Style(styleToApply.Get()));
                         }
                     }
                 }
-                else
+                else if ((SUCCEEDED(WindowsCompareStringOrdinal(actionSentiment.Get(), HStringReference(L"default").Get(), &isSentimentDefault))
+                         && (isSentimentDefault == 0))
+                         || WindowsIsStringEmpty(actionSentiment.Get()))
                 {
                     THROW_IF_FAILED(
                         SetStyleFromResourceDictionary(renderContext, L"Adaptive.Action", buttonFrameworkElement.Get()));
```

Comment:
I assume this should be the most common case? It should be the first conditional in this if/else chain :)